src/Materials/material.ts

```diff
@@ -714,6 +725,44 @@ export class Material implements IAnimatable {
     /** @hidden */
     public _parentContainer: Nullable<AbstractScene> = null;
 
+    /** @hidden */
+    public _dirtyCallbacks: { [code: number]: () => void };
+
+    /** @hidden */
+    public _uniformBufferLayoutBuilt = false;
+
+    protected _eventInfo:
+            MaterialPluginCreated &
+            MaterialPluginDisposed &
+            MaterialPluginHasTexture &
+            MaterialPluginIsReadyForSubMesh &
+            MaterialPluginGetDefineNames &
+            MaterialPluginPrepareEffect &
+            MaterialPluginPrepareDefines &
+            MaterialPluginPrepareUniformBuffer &
+            MaterialPluginBindForSubMesh &
+            MaterialPluginGetAnimatables &
+            MaterialPluginGetActiveTextures &
+            MaterialPluginFillRenderTargetTextures &
+            MaterialPluginHasRenderTargetTextures &
+            MaterialPluginHardBindForSubMesh
+    = { } as any; // will be initialized before each event notification
+
+    /** @hidden */
+    public _callbackPluginEvent: (id: number, info: MaterialPluginGetActiveTextures | MaterialPluginGetAnimatables | MaterialPluginHasTexture | MaterialPluginDisposed | MaterialPluginGetDefineNames | MaterialPluginPrepareEffect | MaterialPluginPrepareUniformBuffer) => void = () => void(0);

```

Comment:
totally unsure about naming here :-)

_pluginCallbackGeneric
_pluginCallbackIsReadyForSubMesh
...
 ?

Not sure event needs to be in but I am stupid with Naming :-)

Reply:
`_pluginCallbackGeneric` or `_pluginCallbackOther` make sense to me, vote your preferences and I'll change it

Reply:
@Popov72  ???